### PR TITLE
Fix - Admin Secrets - "Make All Areas Powered", "Make All Areas Unpowered"

### LINKED ISSUE #14249: "Make all areas unpowered" secret shorts out/overloads random APCs instead of draining power from everything
**Issue Description**:
"Make all areas unpowered" secret shorts out/overloads random APCs instead of draining power from everything.

**What did you expect to happen**: 
For all APCs on the station to get drained of power.

**What happened instead**:
APCs got shorted/overloaded.

**Why is this bad/What are the consequences**:
It makes testing things with power annoying since you can't make all areas unpowered and then repowered them, because the APCs become borked.

**Steps to reproduce the problem**:
1. Open secrets panel
2. Click "Make all areas unpowered"

**When did the problem start happening**:
Likely after this PR: #13117

### PR DESCRIPTION
**Pull request by Kiyahitayika**
## What Does This PR Do
Redo of PR https://github.com/ParadiseSS13/Paradise/pull/14553 due to conflicts with blackbox annhilation.

Tweaks how "Make All Areas Powered" and "Make All Areas Unpowered" work in Admin Secrets commands. 

Fixes #14249, spoke with @SteelSlayer
Continues the work merged in https://github.com/ParadiseSS13/Paradise/pull/13117 and #13712, spoke with @blinkdog

## Why It's Good For The Game
As raised in the issue referenced above, "Make All Areas Powered" does not currently undo the effects of "Make All Areas Unpowered". After speaking with SteelSlayer, it was also brought up that "Make All Areas Unpowered" is a misnomer as it actually shorts out APCs. This PR adds the following features:

**Make All Areas Unpowered** (separate `procs`):
- "Depower all APCs" - sets the charge of all APCs' cells to 0
- "Short out APCs" - shorts out APCs per the current form of "Make All Areas Unpowered"

**Make All Areas Powered:**
- "Power All APCs" - (Power without Repairing) - sets the charge of all functioning APCs' cells to their `maxcharge`.
- "Repair All APCs" - (Repair without Powering) - repairs all APCs to reverse the effects of "Short out APCs"
- "Repair and Power APCs" - (Repair and Power) - repairs all APCs and sets their cells to their `maxcharge`.

Please see the "Images of Changes" video.

## Images of changes
https://imgur.com/a/fPzATEI

## Changelog
:cl:
add: New features to Admin Secrets within "Make all areas powered" and "Make all areas unpowered"
fix: "Make all areas powered" can now undo "Make all areas unpowered"
/:cl:

### CODE REVIEW

File: code/modules/events/apc_short.dm [hunk 1]
```diff
@@ -26,55 +26,95 @@
...
+		var/area/current_area = get_area(A)
+		if((current_area.type in skipped_areas_apc) || !is_station_level(A.z))
+			continue
+		var/obj/item/stock_parts/cell/C = A.get_cell()
+		if(C)
+			C.charge = 0
+		current_area.power_change()
+	log_and_message_admins("Power has been drained from all APCs.")
 
+/proc/power_failure(announce=TRUE)
```
Comment (SabreML) [inline code/modules/events/apc_short.dm:45]:
```suggestion
/proc/power_failure(announce = TRUE)
```
Spacing here

File: code/modules/events/apc_short.dm [hunk 2]
```diff
@@ -26,55 +26,95 @@
...
-
-	// recharge the APCs
-	for(var/thing in GLOB.apcs)
-		var/obj/machinery/power/apc/A = thing
-		if(!is_station_level(A.z))
-			continue
-		var/obj/item/stock_parts/cell/C = A.get_cell()
-		if(C)
-			C.give(C.maxcharge)
+/proc/power_restore(announce=TRUE, var/power_type)
```
Comment (SabreML) [inline code/modules/events/apc_short.dm:75]:
```suggestion
/proc/power_restore(announce = TRUE, var/power_type)
```
Here too

File: code/modules/events/apc_short.dm [hunk 3]
```diff
@@ -26,55 +26,95 @@
...
+	if(power_type == 0)	//Power without Repairing
+		if(announce)
+			GLOB.event_announcement.Announce("All operational APCs on [station_name()] have been fully charged.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
+		var/affected_apc_count = 0
+		for(var/thing in GLOB.apcs)
+			var/obj/machinery/power/apc/A = thing
+			var/area/current_area = get_area(A)
+			if(!is_station_level(A.z))
+				continue
+			if((length(A.wires.cut_wires) == 0) && A.operating && !A.shorted)
```
Comment (SabreML) [inline code/modules/events/apc_short.dm:85]:
```suggestion
			if(!length(A.wires.cut_wires) && A.operating && !A.shorted)
```

**Review by SabreML — APPROVED** [review]

File: code/modules/events/apc_short.dm [hunk 4]
```diff
@@ -26,55 +26,95 @@
...
-		var/obj/machinery/power/apc/A = thing
-		if(!is_station_level(A.z))
-			continue
-		var/obj/item/stock_parts/cell/C = A.get_cell()
-		if(C)
-			C.give(C.maxcharge)
+/proc/power_restore(announce=TRUE, var/power_type)
+	if(power_type == 0)	//Power without Repairing
+		if(announce)
+			GLOB.event_announcement.Announce("All operational APCs on [station_name()] have been fully charged.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
```
Comment (Kiyahitayika) [inline code/modules/events/apc_short.dm:78]: Changing to `GLOB.event_announcement.Announce("All APCs on \the [station_name()]` for grammar along with other instances.

File: code/modules/events/apc_short.dm (hunk 3 above)
Comment (Kiyahitayika) [reply code/modules/events/apc_short.dm:85]: Yes! Totally agree!

File: code/modules/power/apc.dm [hunk 5]
```diff
@@ -1357,6 +1357,19 @@
...
+		toggle_breaker()
+	if(shorted)
+		shorted = FALSE
+
+/obj/machinery/power/apc/proc/recharge_apc()
+	var/obj/item/stock_parts/cell/C = get_cell()
+	if(C)
+		C.charge = C.maxcharge
+
 #undef APC_UPDATE_ICON_COOLDOWN
```
Comment (Kiyahitayika) [inline code/modules/power/apc.dm:1374]: Just noticed the unneeded linebreak on line 1374. Not sure why that's there. Removing.

File: code/modules/events/apc_short.dm (hunk 1 above)
Comment (Kiyahitayika) [reply code/modules/events/apc_short.dm:45]: For some reason, I cannot add my own comments. In checking this file over yet again, I realized there is variable only used once. 

`	var/const/delayed                     = FALSE`
```
/datum/event/apc_short/start()
	power_failure(announce=delayed)
```
I'm going to get rid of it and fix the spacing there as well.

**Review by dearmochi — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**